modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py

```diff
@@ -1397,5 +1398,53 @@ def sort(df, cols, ascending, na_position, **kwargs):
     #    )
 
 
+class TestBadData:
+    bad_for_arrow = {
+        "a": ["a", [[1, 2], [3]], [3, 4]],
+        "b": ["b", [1, 2], [3, 4]],
+        "c": ["1", "2", 3],
+    }
+    bad_for_omnisci = {
+        "b": [[1, 2], [3, 4], [5, 6]],
+        "c": ["1", "2", "3"],
+    }
+    ok_data = {"d": np.arange(3), "e": np.arange(3), "f": np.arange(3)}
+
+    def _get_pyarrow_table(self, obj):
+        if not isinstance(obj, (pandas.DataFrame, pandas.Series)):
+            obj = pandas.DataFrame(obj)
+
+        return pyarrow.Table.from_pandas(obj)
+
+    @pytest.mark.parametrize("data", [bad_for_arrow, bad_for_omnisci])
+    def test_construct(self, data):
+        def applier(df, *args, **kwargs):
+            return repr(df)
+
+        run_and_compare(applier, data=data, force_lazy=False)
+
+    def test_from_arrow(self):
+        at = self._get_pyarrow_table(self.bad_for_omnisci)
+        md_df, pd_df = pd.utils.from_arrow(at), pandas.DataFrame(self.bad_for_omnisci)
+        # force materialization
```

Comment:
We don't need to force materialization, comparison function does it already.

Reply:
yes, the comparison function materializes frames, but it does not trigger any computation to execute.

If we'll create a new modin frame and right after that do `._to_pandas()` then we'll just return our pandas frame from partition and will not hit the execute stage
https://github.com/modin-project/modin/blob/dd691cd0d4bb5403b38242563ed8b6ddb316734a/modin/experimental/engines/omnisci_on_ray/frame/data.py#L905-L907

Reply:
If we have frames materialized and execution not triggered then we just have nothing to execute. `FrameNode` is always a computation tree list.

Reply:
yeah, but what we want to do is trigger the execution stage after constructing frame to check if we build it correctly

Reply:
I'm not sure what you mean by triggering the execution stage. We cannot trigger any execution for `FrameNode`. There is just nothing to execute.

Reply:
if we do
```python
pd.DataFrame(unsupported_data)._to_pandas()
```
then we'll just put pandas DataFrame into partition and right after that getting it from there, so execution stage was not triggered and that code will work even at the current master 

if we do
```python
df = pd.DataFrame(unsupported_data)
repr(df)
df._to_pandas()
```
then `repr` will trigger index accessing, frame slicing etc... , so some computation is required and we will potentially hit execution on OmniSci (which we don't want to in case of unsupported data)

Reply:
I see. `repr` triggers some slicing but we execute it on an arrow table using `pyarrow`. If the frame was created using `from_pandas` then `repr` might trigger execution on OmniSci (BTW we should use `pyarrow` option here as well since we convert `pandas.DataFrame` to `pyarrow.Table` anyway)